hnn_core/feed.py

```diff
@@ -5,219 +5,10 @@
 #          Christopher Bailey <bailey.cj@gmail.com>
 
 import numpy as np
-from neuron import h
 
 from .utils import check_random_state
 
 
-class Rhythmic:
-    """External inputs
-
-    Parameters
-    ----------
-    loc : str ('prox' or 'dist')
-        The location of the rhythmic input
-    params : dict
-        The params dictionary. For compatibility
-        with GUI.
-
-    Attributes
-    ----------
-    loc : str ('prox' or 'dist')
-        The location of the rhythmic input
-    t_input : array
-        The spike times.
-    """
-    def __init__(self, loc, params=None):
-
-        self.loc = loc
-        self.t_input = None
-
-        if params is not None:
-            self.set_times(tstart=params[f't0_input_{loc}'],
-                           tstop=params[f'tstop_input_{loc}'],
-                           burst_freq=params[f'f_input_{loc}'],
-                           burst_std=params[f'f_stdev_{loc}'],
-                           burst_n_spikes=params[f'events_per_cycle_{loc}'],
-                           n_bursts=params[f'repeats_{loc}'])
-            self.set_prox_connections_layer2(
-                g_pyr_ampa=params['input_prox_A_weight_L2Pyr_ampa'],
-                g_pyr_nmda=params['input_prox_A_weight_L2Pyr_nmda'],
-                g_basket_ampa=params['input_prox_A_weight_L2Basket_ampa'],
-                g_basket_nmda=params['input_prox_A_weight_L2Basket_nmda'],
-                delay=params['input_prox_A_delay_L2'])
-            self.set_prox_connections_layer5(
-                g_pyr_ampa=params['input_prox_A_weight_L5Pyr_ampa'],
-                g_pyr_nmda=params['input_prox_A_weight_L5Pyr_nmda'],
-                g_basket_ampa=params['input_prox_A_weight_L5Basket_ampa'],
-                g_basket_nmda=params['input_prox_A_weight_L5Basket_nmda'],
-                delay=params['input_prox_A_delay_L5'])
-            self.set_dist_connections_layer2(
-                g_pyr_ampa=params['input_dist_A_weight_L2Pyr_ampa'],
-                g_pyr_nmda=params['input_dist_A_weight_L2Pyr_nmda'],
-                g_basket_ampa=params['input_dist_A_weight_L2Basket_ampa'],
-                g_basket_nmda=params['input_dist_A_weight_L2Basket_nmda'],
-                delay=params['input_dist_A_delay_L5'])
-            self.set_dist_connections_layer5(
-                g_pyr_ampa=params['input_dist_A_weight_L5Pyr_ampa'],
-                g_pyr_nmda=params['input_dist_A_weight_L5Pyr_nmda'],
-                delay=params['input_dist_A_delay_L2'])
-
-    def set_times(self, tstart, tstop, tstart_dev, burst_freq=10, burst_std=20,
-                  burst_n_spikes=2, n_bursts=10, randomize_tstart=False,
-                  random_state1=None, random_state2=None):
```

Comment:
not sure what you were attempting but I think we should keep this method as a function. Let's talk more tomorrow. Maybe I'm overlooking some challenge